hyperon_das/index.py

```diff
@@ -0,0 +1,30 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class IndexField:
+    collection: str
```

Comment:
Our discussion is running in circles.
We can't have DB-specific details in DAS. So we can't reference "collection", "direction" and "conditionals"